packages/zipkin/src/tracer/index.js

```diff
@@ -151,15 +151,20 @@ class Tracer {
   }
 
   recordAnnotation(annotation) {
+    let shouldRecord = this.id.flags === 1;
```

Comment:
@adriancole I am not entirely sure why we were not checking the debug flags for recording spans. Any input?

Reply:
debug needs to end up as span.debug when reporting. In order for the trace to be consistently debug (ex when propagating downstream and at collector tier), it needs to be carried somehow (flag or otherwise in the context), and also written as span.debug=true in json (note we don't ever write span.debug=false)